lib/libzpool/kernel.c

```diff
@@ -1090,64 +1089,47 @@ zvol_rename_minors(spa_t *spa, const char *oldname, const char *newname,
 int
 zfs_file_open(const char *path, int flags, int mode, zfs_file_t **fpp)
 {
-	int fd = -1;
-	int dump_fd = -1;
-	int err;
-	int old_umask = 0;
-	zfs_file_t *fp;
-	struct stat64 st;
-
-	if (!(flags & O_CREAT) && stat64(path, &st) == -1)
-		return (errno);
-
-	if (!(flags & O_CREAT) && S_ISBLK(st.st_mode))
-		flags |= O_DIRECT;
-
-	if (flags & O_CREAT)
-		old_umask = umask(0);
-
-	fd = open64(path, flags, mode);
-	if (fd == -1)
-		return (errno);
-
-	if (flags & O_CREAT)
-		(void) umask(old_umask);
-
-	if (vn_dumpdir != NULL) {
-		char *dumppath = umem_zalloc(MAXPATHLEN, UMEM_NOFAIL);
-		char *inpath = basename((char *)(uintptr_t)path);
-
-		(void) snprintf(dumppath, MAXPATHLEN,
-		    "%s/%s", vn_dumpdir, inpath);
-		dump_fd = open64(dumppath, O_CREAT | O_WRONLY, 0666);
-		umem_free(dumppath, MAXPATHLEN);
-		if (dump_fd == -1) {
-			err = errno;
-			close(fd);
-			return (err);
-		}
-	} else {
-		dump_fd = -1;
-	}
-
-	(void) fcntl(fd, F_SETFD, FD_CLOEXEC);
-
-	fp = umem_zalloc(sizeof (zfs_file_t), UMEM_NOFAIL);
-	fp->f_fd = fd;
-	fp->f_dump_fd = dump_fd;
-	*fpp = fp;
-
-	return (0);
```

Comment:
Hoping wosix.h will make all this working without much change, and any additional fixes to go into our posix.c